include/cuda/std/stream

```diff
@@ -0,0 +1,156 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA_STREAM
+#define _CUDA_STREAM
+
+#include "cstddef"
+
+#include "detail/__config"
+#include "detail/__pragma_push"
+#include "version"
+
+#include <cuda_runtime_api.h>
+#include <memory>
+#include <stdexcept>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+/**
+ * @brief A non-owning wrapper for a `cudaStream_t`.
+ *
+ */
+class stream_view {
+
+  /**
+   * @brief Constructs a `stream_view` of the "default" CUDA stream.
+   *
+   * For behavior of the default stream,
+   * see https://docs.nvidia.com/cuda/cuda-runtime-api/stream-sync-behavior.html
+   *
+   */
+  constexpr stream_view() = default;
+
+  /**
+   * @brief Constructs a `stream_view` from a `cudaStream_t` handle
+   *
+   * @note: It is the callers responsibilty to ensure the `stream_view` does not
+   * outlive the stream identified by the `cudaStream_t` handle.
+   *
+   */
+  explicit constexpr stream_view(cudaStream_t stream) : __stream{stream} {}
+
+  /// Disallow construction from an `int`, e.g., `0`.
+  stream_view(int) = delete;
+
+  /// Disallow construction from `nullptr`
+  stream_view(std::nullptr_t) = delete;
+
+  /// Returns the wrapped `cudaStream_t` handle
+  constexpr cudaStream_t value() const noexcept { return __stream; }
+
+  /// Explicit conversion to `cudaStream_t` returns the wrapped `cudaStream_t`
+  /// handle
+  explicit constexpr operator cudaStream_t const noexcept { return value(); }
```

Comment:
Why explicit? I thought the original point of even having stream view was to have a stronger cudaStream_t, which would resolve ambiguity between `int`, `nullptr_t` and `cudaStream_t` - I think that even with implict conversion to cudaStream_t this is satisified and it does not require any code changes when cudaStream_t is replaced with stream_view.
Moreover, explicit conversion results in awkward syntax - to the point where most users would probably just use `value()`.

Reply:
I think in RMM we just have `value()`, right?

Reply:
Implicit conversions almost always cause more problems than they solve. I'd rather remove the conversion operator all together than make it implicit.

Reply:
Removed the conversion operator to `cudaStream_t` in favor of using `get()`.